config/templates/stack-template.json

```diff
@@ -87,9 +87,44 @@
           {{end}}
         }
       },
-      "DependsOn" : ["AutoScaleController"]
+      "Metadata" : {
```

Comment:
Thanks for your comment!
Regarding the mental effort, your original decision sounds very reasonable to me now.
On the other hand, I'd also like only relevant configsets to be included in the metadata of a AWS resource like an ASG for workers and an ASG for controllers, respectively.
So, would you think it is not the best but acceptable/slightly better to extract each configset to a golang subtemplate like:

```yaml
{{define "cfn-init-configset-aws-environment"}}
{
  "commands": {
  "write-environment": {
  "command": { "Fn::Join" : ["", [ "echo '",
  {{range $variable, $function := .Experimental.AwsEnvironment.Environment}}
  "{{$variable}}=", {{$function}} , "\n",
  {{end}}
  "' > /etc/aws-environment" ] ] }
  }
}
{{end}}
{{define "cfn-init-configset-etcd-client"}}
...
{{end}}
{{define "cfn-init-configset-etcd-server"}}
...
{{end}}
```